TANMEDIAGROUP/Syndex-Fn
# Ch(V1.0):initial release

### What does this PR do?

This PR adds several components and functionalities to the existing codebase, including landing, login, sign up, and 404 pages, as well as Storybook documentation and Vitest for testing.

### Description of Task to be completed?

The tasks completed in this PR include the development of the following components and features:

#### tasks

- [x] Created a visually appealing and informative landing page
- [x] Implemented a secure and user-friendly login page
- [x] Developed a sign-up page with form validation and error handling
- [x] Added a custom 404 page to handle non-existent routes
- [x] Integrated Storybook for component documentation and development
- [x] Set up Vitest for unit and integration testing
- [x] Wrote comprehensive tests for the new components and features
- [x] Ensured code quality and adherence to best practices

### How to test

- Step 1: Clone the repository and navigate to the project directory
- Step 2: Install dependencies by running `npm install`
- Step 3: Start the development server with `npm run dev`
- Step 4: Open your web browser and navigate to `http://localhost:3000`
- Step 5: Explore the new pages (landing, login, sign up, and 404)
- Step 6: Run Storybook with `npm run storybook` to view component documentation
- Step 7: Run tests with `npm test` to ensure all tests pass

### Any background context you want to provide?

This PR is part of a larger effort to enhance the user experience and provide a more robust and feature-rich application. The landing page serves as the primary entry point for new users, while the login and sign-up pages enable existing and new users to access the application's core features. The 404 page ensures a graceful handling of non-existent routes. Storybook documentation and Vitest testing contribute to better maintainability and reliability of the codebase.

### What are the relevant Github Issues if any?

This PR addresses the following Github Issues:
- Testing with Vitest (N/A not raised)
